File: app/ios/objc/Khipu Inside Demo/Khipu Inside Demo/ViewController.m
//
//  ViewController.m
//  Khipu Inside Demo
//
//  Created by Emilio Davis on 7/4/18.
//  Copyright © 2018 Emilio Davis. All rights reserved.
//

#import "ViewController.h"
#import <khenshin/khenshin.h>

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextField *paymentId;

@end

@implementation ViewController

- (IBAction)doPay:(UIButton *)sender {
    [KhenshinInterface startEngineWithPaymentExternalId:[[self paymentId] text]
                                         userIdentifier:@""
                                      isExternalPayment:YES
                                                success:^(NSURL *returnURL){
        NSLog(@"SUCCESS");
    }
                                                failure:^(NSURL *returnURL){
        NSLog(@"FAILURE");
    }
                                               animated:NO];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: app/ios/swift/Khipu Inside Demo/Khipu Inside Demo/ViewController.swift
//
//  ViewController.swift
//  Khipu Inside Demo
//
//  Created by Emilio Davis on 7/4/18.
//  Copyright © 2018 Emilio Davis. All rights reserved.
//

import UIKit
import khenshin

class ViewController: UIViewController {

    @IBOutlet weak var paymentId: UITextField!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func doPay(_ sender: UIButton) {
        
        KhenshinInterface.startEngine(withPaymentExternalId: paymentId.text, userIdentifier: "", isExternalPayment: true, success: { (exitURL: URL?) in
            NSLog("SUCCESS")
        }, failure: { (exitURL: URL?) in
            NSLog("FAILURE")
        }, animated: false)
        
    }
    
}

